### ISSUE TITLE
error_class=Elasticsearch::Transport::Transport::Errors::BadRequest error="[400] {\"error\":\"ElasticsearchParseException[Failed to derive xcontent]\",\"status\":400}

### ISSUE THREAD

**Issue by avikam** [uken/fluent-plugin-elasticsearch#204]
(same issue as here: https://github.com/uken/fluent-plugin-elasticsearch/issues/187), diagnosed differently.

#### Problem

fluentd root agent has a lifecycle method that is executed on shutdown. It calls methods dynamically using [send Ruby's keyword](http://ruby-doc.org/core-2.3.1/Object.html#method-i-send) .
relevant fluentd code [here](https://github.com/fluent/fluentd/blob/master/lib/fluent/root_agent.rb#L174) and [here](https://github.com/fluent/fluentd/blob/master/lib/fluent/root_agent.rb#L200)
However this plugin is [overriding send method](https://github.com/uken/fluent-plugin-elasticsearch/blob/0e455efa82/lib/fluent/plugin/out_elasticsearch.rb#L331), which leads to a weird and buggy behaviour in which td-agent is sending the following raw data to elasticsearch:
 
```
POST http://elasticsearch/_bulk 
User-Agent:      Faraday v0.9.2
Host:            elasticsearch
Content-Length:  10
"stopped?"
```

and gets this response:
```
400 Bad Request
Access-Control-Allow-Origin:  *
Content-Type:                 application/json; charset=UTF-8
Server:                       Jetty(8.1.12.v20130726)
Content-Length:               135
Connection:                   keep-alive
{
    "error": "ElasticsearchParseException[Failed to derive xcontent from org.elasticsearch.common.bytes.BytesArray@68b8132f]",
    "status": 400
}
```

Here is the log:
```
BadRequest error="[400] {\"error\":\"ElasticsearchParseException[Failed to derive xcontent from org.elasticsearch.common.bytes.BytesArray@fcfd7d1d]\",\"status\":400}"
/td-agent.log- 2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/elasticsearch-transport-1.0.18/lib/elasticsearch/transport/transport/base.rb:201:in `__raise_transport_error'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/elasticsearch-transport-1.0.18/lib/elasticsearch/transport/transport/base.rb:312:in `perform_request'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/elasticsearch-transport-1.0.18/lib/elasticsearch/transport/transport/http/faraday.rb:20:in `perform_request'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/elasticsearch-transport-1.0.18/lib/elasticsearch/transport/client.rb:128:in `perform_request'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/elasticsearch-api-1.0.18/lib/elasticsearch/api/actions/bulk.rb:90:in `bulk'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.7.0/lib/fluent/plugin/out_elasticsearch.rb:334:in `send'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.8/lib/fluent/root_agent.rb:174:in `block (2 levels) in shutdown'
td-agent.log-  2016-10-19 20:58:31 +0000 [warn]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.14.8/lib/fluent/root_agent.rb:127:in `block (2 levels) in lifecycle'
```

I think it can be solved by changing the method's name

**Comment by pitr** [comment]
sounds good, can you open a PR?

**Comment by pitr** [comment]
fixed by #206
